### PR TITLE
Fix more diagnostic wording for style; NFC

### PR DESCRIPTION
**Pull request by AaronBallman**
This tries to fix all of the places where a diagnostic message starts with a capital letter (other than acroynyms or proper nouns) or ends with punctuation (other than a question mark).

This is in support of a planned change to tablegen to start diagnosing incorrect diagnostic message styles.

### CODE REVIEW

**Comment by AaronBallman** [comment]
All of the issues were found using some new tooling I've not yet put up for review. The changes were largely mechanical (uppercase->lowercase, remove trailing punctuation), but there were some surgical fixes to grammar, punctuation, etc. Also, if the diagnostic was not emitted anywhere, it was removed entirely.

File: clang/include/clang/Basic/DiagnosticFrontendKinds.td [hunk 1]
```diff
@@ -337,7 +337,7 @@ def warn_atomic_op_oversized : Warning<
 InGroup<AtomicAlignment>;
 
 def warn_sync_op_misaligned : Warning<
-  "__sync builtin operation MUST have natural alignment (consider using __atomic).">,
+  "__sync builtin operation MUST have natural alignment (consider using __atomic)">,
```
Comment (Fznamznon) [inline clang/include/clang/Basic/DiagnosticFrontendKinds.td:340]: So, it is not allowed to capitalize a message, but a whole word written (`MUST` in this example) with uppercase is allowed? Isn't it too much?

**Comment by AaronBallman** [comment]
Despite not showing up in the list of precommit CI checks for some reason, this did pass testing: https://buildkite.com/llvm-project/github-pull-requests/builds/66650

**Review by cyndyishida — APPROVED** [review]
the InstallAPI changes LGTM, thanks!

File: clang/include/clang/Basic/DiagnosticSemaKinds.td [hunk 2]
```diff
@@ -4526,7 +4526,7 @@ def err_objc_attr_typedef_not_void_pointer : Error<
 def err_objc_cf_bridged_not_interface : Error<
   "CF object of type %0 is bridged to %1, which is not an Objective-C class">;
 def err_objc_ns_bridged_invalid_cfobject : Error<
-  "ObjectiveC object of type %0 is bridged to %1, which is not valid CF object">;
+  "Objective-C object of type %0 is bridged to %1, which is not valid CF object">;
```
Comment (yronglin) [inline clang/include/clang/Basic/DiagnosticSemaKinds.td:4529]: I've found `ObjectiveC` still exists in elsewhere:
https://github.com/llvm/llvm-project/blob/5a81db311cf8e15b7c6a33100bb5c1fb256754ce/clang/include/clang/Basic/DiagnosticParseKinds.td#L1120
https://github.com/llvm/llvm-project/blob/5a81db311cf8e15b7c6a33100bb5c1fb256754ce/clang/include/clang/Basic/DiagnosticSemaKinds.td#L4529

Comment (AaronBallman) [reply clang/include/clang/Basic/DiagnosticSemaKinds.td:4529]: Good catch, I've fixed it up

File: clang/include/clang/Basic/DiagnosticSemaKinds.td [hunk 3]
```diff
@@ -10362,12 +10362,12 @@ def err_shufflevector_argument_too_large : Error<
   "index for __builtin_shufflevector must be less than the total number "
   "of vector elements">;
 def err_shufflevector_minus_one_is_undefined_behavior_constexpr : Error<
-  "index for __builtin_shufflevector not within the bounds of the input vectors; index of -1 found at position %0 not permitted in a constexpr context.">;
+  "index for __builtin_shufflevector not within the bounds of the input vectors; index of -1 found at position %0 not permitted in a constexpr context">;
```
Comment (Endilll) [inline clang/include/clang/Basic/DiagnosticSemaKinds.td:10365]:
```suggestion
  "index for __builtin_shufflevector not within the bounds of the input vectors; index of -1 found at position %0 is not permitted in a constexpr context">;
```

**Review by Endilll — APPROVED** [review]

File: clang/include/clang/Basic/DiagnosticSemaKinds.td (hunk 3 above)
Comment (AaronBallman) [reply clang/include/clang/Basic/DiagnosticSemaKinds.td]: SGTM, done!

**Pull request merged**